managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/ManagedCursorImpl.java

```diff
@@ -3051,7 +3051,15 @@ public void markDeleteComplete(Object ctx) {
 
             @Override
             public void markDeleteFailed(ManagedLedgerException exception, Object ctx) {
-                log.warn("[{}][{}] Failed to flush mark-delete position", ledger.getName(), name, exception);
+                //They throw an IllegalArgumentException when the mark deletes a position that has already been deleted.
+                //See this#setAcknowledgedPosition
+                if (exception.getCause() instanceof IllegalArgumentException) {
```

Comment:
For ManagedCursorImpl#setAcknowledgedPosition, could we add a new exception :
```
if (newMarkDeletePosition.compareTo(markDeletePosition) < 0) {
    throw new AlreadySetMarkDeleteException("Mark deleting an already mark-deleted position");
}
```
or print log here?
```
if (newMarkDeletePosition.compareTo(markDeletePosition) < 0) {
    log.warn("Mark deleting an already mark-deleted position : {}", newMarkDeletePosition);
    return markDeletePosition;
}
```
@mattisonchao @codelipenghui @merlimat

Reply:
@Technoboy-  

I think we need to find all the logic about this and reconsider if we need to throw an exception.